### DISCUSSION TITLE
Define basis for higher spin chain using z-projection number and evaluating expectation value using it.

### DISCUSSION THREAD

**Discussion by thefacemyer** [QuSpin/QuSpin#536] [Q&A] [answered]
Hello, I'm relatively new to coding and QuSpin (perhaps a bad combination). Can someone help me understand how I can find the expectation value for the Heisenberg Hamiltonian with a user-defined state like |3/2, 3/2, -3/2>? So, for example, how can I use QuSpin to evaluate <3/2, 3/2, -3/2|H|3/2, 3/2, -3/2>?

**Comment by weinbe58** [comment, marked as answer]
There are many ways o do this calculation. I Would suggest you first look at some of the examples specifically [example 0](http://weinbe58.github.io/QuSpin/examples/example0.html#example0-label). To study higher spin chain you simply use add an optional argument: `S="3/2"` when you construct the basis. 

The thing you want to compute can be obtained directly from the hamiltonian object so long as you get the index of the fock state you are interested in. 

To do this you need to first take your fock state and represent it as a bosonic state where the boson occupation of each site is given by:  Sz + S where S is 3/2 in your example: so your bosonic state would be | 3 3 0 >. Next with that state obtained you can use the `basis` object to get the index by using the [index function](http://weinbe58.github.io/QuSpin/generated/quspin.basis.spin_basis_1d.html#quspin.basis.spin_basis_1d.index). 

so what this amounts to is first constructing your hamiltonian object `H` and then using the following code:

```python
state = "330"
i = H.basis.index(state)
H_me = H[i,i]
```

**Reply by thefacemyer** [reply]
Thank you so much. I have gone through the first couple of examples, and they have been very useful. I found in the documentation that we can directly evaluate the expectation values using expt_value and then calling a state "V." I'm assuming that this can be done by defining V in terms of the bosonic state you mention, but my question, I think, had more to do with whether we can represent our states in the spin-projection basis.

**Reply by weinbe58** [reply]
I am not sure what you mean by "spin-projection basis". By default, the Hamiltonian is written in a basis of Fock states parameterized with the local Sz projection on each site. You can change the basis if you have a method to generate that transformation as a matrix in python.

**Reply by thefacemyer** [reply]
Sorry for being unclear. I am wondering if QuSpin has some way of say H_expt = H.expt_value(V,time=0,diagonal=False,check=True), where V is explicitly written in Sz projection basis. So, V = '|3/2, 3/2, -3/2>', rather than needing the Fock states. The reason is I'm not usually interested in fermionic or bosonic transformations of the chain, just a brute force diagonalization of the spins in the Heisenberg model. Maybe my confusion stems from my naivete of how QuSpin handles the model we plug in when I follow example0. I don't see anything that explicitly moves from the Sz basis to a number basis.

**Reply by thefacemyer** [reply]
Okay, I printed out the bases and see the default bases you talked about. These Fock states are already in the form of |3,3,0>, so I went ahead and use the code you provided, and it returned an error: IndexError: invalid number of indices, hamiltonian must be indexed with three indices [time,row,col]. I'm using the hamiltonian in example0, but changed to it account for the higher spin.
